lib/sidekiq/pool/cli.rb

```diff
@@ -49,8 +49,14 @@ def start_new_pool
         @types = @settings[:workers]
         @types.each do |type|
           type[:amount].times do
-            sleep @fork_wait || DEFAULT_FORK_WAIT
-            add_child(type[:command])
+            fork_wait = @fork_wait || DEFAULT_FORK_WAIT
+            times_to_wait_for_alive ||= (fork_wait * 3)
+            pid = add_child(type[:command])
+
+            until alive?(pid)
```

Comment:
I think `alive?` is not enough because it just checks if process is alive. I suggest just forking children as before and then have method checking if process is loaded (procline changes).